testing/test_driver.py

```diff
@@ -180,3 +181,70 @@ def test_pass(driver_kwargs):
     """
     )
     testdir.quick_qa("--driver", "Firefox", file_test, passed=1)
+
+
+def test_driver_retry_pass(testdir, mocker):
+    mock_retrying = mocker.spy(pytest_selenium.pytest_selenium, "Retrying")
+    file_test = testdir.makepyfile(
+        """
+        import pytest
+
+        class MockDriver():
+            def __init__(*args, **kwargs):
+                pass
+            def quit(*args, **kwargs):
+                pass
+
+        @pytest.fixture(scope="session")
+        def driver_class():
+            return MockDriver
```

Comment:
I tried using the mocker to avoid having these, but couldn't figure out how.

I tried:

```python
mocker.patch("pytest_selenium.pytest_selenium.webdriver")
```

and `"pytest_selenium.pytest_selenium.webdriver.Firefox"`

but couldn't get it to work 🤷‍♂️

Reply:
I don't have too much experience with using `mock` but I think a better test is to use `Remote` webdriver, and then  mock the `RemoteConnection` class to throw an exception. This would be more realistic than having Firefox fail. Either way, focus on remote connection failing.

Reply:
That's genius! 

But how do I accomplish that?

I tried
```python
    mocker.patch(
        "pytest_selenium.pytest_selenium.webdriver.remote.remote_connection.RemoteConnection",
        side_effect=Exception("Connection Error"),
    )
```

But it doesn't work. With that I mean, I don't get any syntax errors - but the test doesn't produce the expected outcome (the exception isn't raised.

Reply:
I'm struggling to get something to work.

I managed to mock the `execute` method on `RemoteConnection`, but that slowed down the tests (from ~1s to ~6s).

Maybe @nicoddemus can share some wisdom?

Reply:
Think I solved it. 👍

Reply:
Sorry I didn't get here sooner.

Was the fix moving construction of `driver_class` inside the `with` statement?

Reply:
No worries @nicoddemus Sorry I squashed the commits and hid the history.

But no, the fix was doing the patching in the proper places.